include/nighthawk/common/statistic.h

```diff
@@ -115,6 +116,23 @@ class Statistic : Envoy::NonCopyable {
    * @param id The id that should be set for the Statistic instance.
    */
   virtual void setId(absl::string_view id) PURE;
+
+  /**
+   * Build a string representation of this Statistic instance.
+   *
+   * @return absl::StatusOr<std::unique_ptr<std::istream>> Status or a stream that will yield
+   * a serialized representation of this Statistic instance.
+   */
+  virtual absl::StatusOr<std::unique_ptr<std::istream>> serializeNative() const PURE;
```

Comment:
Curious - what advantage do we gain from serializing into a `std::istream` rather than `std::string`? Is it something that helps the reader / user? Since it doesn't seem to be helping writer much.

Reply:
I agree that for this implementation for HdrHistogram we could use `std::string`. 
I was hoping that using streams here leaves some wiggle room to move to chunked reading and writing (and maybe use a stream wrapper to apply compression) if at some point another stats implementation implements this, and needs to write large volumes of data.
At the extreme, I could see that becoming useful if we would want to serialize a raw latency stream at some point.

Reply:
Thank you for explaining.